src/main/kotlin/NaiveBayes.kt

```diff
@@ -1,45 +1,37 @@
 import com.beust.klaxon.JsonObject
 import com.beust.klaxon.Parser
 import java.io.File
-import java.util.*
 
-fun extractTweetsFromJSON (path: String): List<String> {
-    val tweets = mutableListOf<String>()
-    val parser = Parser()
-    File(path).forEachLine { line -> tweets.add((parser.parse(StringBuilder(line)) as JsonObject).string("text").toString()) }
-    return Collections.unmodifiableList(tweets)
+fun extractTweetsFromJSON(path: String): List<String> {
+    val parser = Parser.default()
+    return File(path).readLines().map { (parser.parse(StringBuilder(it)) as JsonObject).string("text") ?: "null" }
 }
 
 data class TweetRecord(val tweetTokens: List<String>, val label: Int)
 
-fun trainTestSplit (dataset: List<TweetRecord>, testSize: Double = 0.2): Pair<List<TweetRecord>,List<TweetRecord>>{
+fun trainTestSplit(dataset: List<TweetRecord>, testSize: Double = 0.2): Pair<List<TweetRecord>, List<TweetRecord>> {
     val shuffledDataset = dataset.shuffled()
-    val trainSetSize: Int = (shuffledDataset.size*(1 - testSize)).toInt()
+    val trainSetSize: Int = (shuffledDataset.size * (1 - testSize)).toInt()
     val trainSet = shuffledDataset.take(trainSetSize)
-
-    val testSetSize: Int = (shuffledDataset.size*testSize).toInt()
-    val testSet = shuffledDataset.takeLast(testSetSize)
-
-    return Pair(trainSet, testSet)
+    val testSet = shuffledDataset.drop(trainSetSize)
+    return trainSet to testSet
```

Comment:
This may be a personal preference but I actually find Pair(a, b) to be more readable than a to b